### ISSUE TITLE
get 'path' fails when path contains invalid UTF-8 characters

### ISSUE THREAD

**Issue by unikitty37** [rspec/rspec-rails#2150]
## What Ruby, Rails and RSpec versions are you using?

Ruby version: ruby 2.3.3p222 (2016-11-21 revision 56859) [x86_64-darwin18]
Rails version: 5.0.6
RSpec version: 3.7

## Observed behaviour

This test fails

```ruby
require 'rails_helper'

describe Rack::Attack do
  include Rack::Test::Methods

  let(:blocked) { Rack::Utils::SYMBOL_TO_STATUS_CODE[:forbidden] }

  def app
    Rails.application
  end

  describe 'blocking' do
    it 'blocks URLs with invalid UTF-8 characters' do
      get "/?q=a\xff"
      expect(last_response.status).to eq(blocked)
    end
  end
end
```

with the following output

```
URI::InvalidURIError: URI must be ascii only "/?q=a\xFF"

  0) Rack::Attack vulnerability probes blocks URLs with invalid UTF-8 characters
     Failure/Error: get "/?q=a\xff"

     URI::InvalidURIError:
       URI must be ascii only "/?q=a\xFF"
     # ./spec/initializers/rack_attack_spec.rb:60:in `block (3 levels) in <top (required)>'
```

## Expected behaviour

The URI is not validated (as attackers will not be passing *their* URIs through validators!) but is accessed, and blocked by `Rack::Attack`.

## Can you provide an example app?

Unfortunately not — `rails _5.0.6_ new issue` created an app using rails 5.2.3, and changing the version in the Gemfile to 5.0.6 and running `bundle update` produced this:

```
└─▷ bundle exec rails g rspec:install
/Users/me/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/railties-5.0.6/lib/rails/railtie/configuration.rb:95:in `method_missing': undefined method `load_defaults' for #<Rails::Application::Configuration:0x007f8b67cd50d8> (NoMethodError)
	from /Users/me/issue/config/application.rb:12:in `<class:Application>'
	from /Users/me/issue/config/application.rb:10:in `<module:Issue>'
	from /Users/me/issue/config/application.rb:9:in `<top (required)>'
	from /Users/me/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/spring-2.0.2/lib/spring/application.rb:92:in `require'
	from /Users/me/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/spring-2.0.2/lib/spring/application.rb:92:in `preload'
	from /Users/me/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/spring-2.0.2/lib/spring/application.rb:153:in `serve'
	from /Users/me/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/spring-2.0.2/lib/spring/application.rb:141:in `block in run'
	from /Users/me/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/spring-2.0.2/lib/spring/application.rb:135:in `loop'
	from /Users/me/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/spring-2.0.2/lib/spring/application.rb:135:in `run'
	from /Users/me/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/spring-2.0.2/lib/spring/application/boot.rb:19:in `<top (required)>'
	from /Users/me/.rbenv/versions/2.3.3/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /Users/me/.rbenv/versions/2.3.3/lib/ruby/2.3.0/rubygems/core_ext/kernel_require.rb:55:in `require'
	from -e:1:in `<main>'
```

If I try this with Rails 5.2.3, `bundle exec rails g rspec:install` hangs and never prints any output, nor does it respond to presses of ^C — I need to kill it manually.

**Comment by unikitty37** [comment]
Ah, apologies — this looks to be a rack-test issue, not rspec rails.
